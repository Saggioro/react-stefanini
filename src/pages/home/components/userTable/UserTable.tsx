import React, { FunctionComponent, useState } from "react";
import Input from "../../../../components/input/Input";
import { useAppSelector } from "../../../../hooks/LocalReduxThunk";

import { Container, Table } from "./UserTable.styled";

interface IUserTable {}

const UserTable: FunctionComponent<IUserTable> = ({ ...props }: IUserTable) => {
  const users = useAppSelector((state) => state.usersReducer.users);
  const [filter, setFilter] = useState<string>("");
  const filterUsers = (word: string) => {
    if (!word || word.trim().length < 3) {
      return users;
    }
    return users.filter((user) => user.name.search(word) >= 0);
  };
  return (
    <Container>
      <h2>Users</h2>
      <Input
        value={filter}
        name="filter"
        label="Filter"
        onChange={(e: any) => setFilter(e.target.value)}
      />
      <Table>
        <thead>
          <tr>
            <th>Login</th>
            <th>Password</th>
          </tr>
        </thead>
        <tbody>
          {filterUsers(filter).map((user) => (
            <tr key={user.name}>
              <td>{user.name}</td>
              <td>{user.password}</td>
            </tr>
          ))}
        </tbody>
      </Table>
    </Container>
  );
};

export default UserTable;
