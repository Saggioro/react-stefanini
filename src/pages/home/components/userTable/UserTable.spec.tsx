/* eslint-disable testing-library/no-node-access */
/* eslint-disable testing-library/no-container */
/* eslint-disable testing-library/prefer-screen-queries */
/* eslint-disable testing-library/no-debugging-utils */
import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import UserTable from "./UserTable";
jest.mock("react-redux", () => {
  return {
    useDispatch() {
      return {
        dispatch: (user: any) => {
          user = undefined;
        },
      };
    },
  };
});

jest.mock("../../../../hooks/LocalReduxThunk", () => {
  return {
    useAppSelector() {
      return [{
        name: "admin teste",
        password: "123",
      }];
    },
  };
});

test("UserTable should render correctly", () => {
  const component = renderer.create(<UserTable />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("UserTable should render users names and passwords", () => {
    render(<UserTable />);
  
    expect(screen.getByText("admin teste")).toBeInTheDocument();
    expect(screen.getByText("123")).toBeInTheDocument();
  });
  test("UserTable should filter users names", () => {
    const {container} = render(<UserTable />);
    const input = container.querySelector(`input[name="filter"]`) as Element;
    fireEvent.change(input, { target : {value: 'admin'}});
    expect(screen.getByText("admin teste")).toBeInTheDocument();
    expect(screen.getByText("123")).toBeInTheDocument();
  });
  
