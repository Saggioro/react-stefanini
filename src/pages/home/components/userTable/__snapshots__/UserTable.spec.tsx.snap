// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTable should render correctly 1`] = `
<div
  className="sc-eCYdqJ cKldaa"
>
  <h2>
    Users
  </h2>
  <div
    className="sc-bczRLJ goEvZo"
  >
    <label
      className="sc-dkzDqf hbcVSe"
    >
      Filter
    </label>
    <input
      className="sc-gsnTZi flIaCB"
      name="filter"
      onChange={[Function]}
      type="text"
      value=""
    />
    <span
      className="sc-hKMtZM ftuqSP"
    />
  </div>
  <table
    className="sc-jSMfEi dPlBUE"
  >
    <thead>
      <tr>
        <th>
          Login
        </th>
        <th>
          Password
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          admin teste
        </td>
        <td>
          123
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
